tools/scalajsenv.js

```diff
@@ -180,11 +180,11 @@ ScalaJS.isScalaJSObject = function(obj) {
 //!if asInstanceOfs != Unchecked
 ScalaJS.throwClassCastException = function(instance, classFullName) {
 //!if asInstanceOfs == Compliant
-  throw new ScalaJS.c.jl_ClassCastException().init___T(
+  throw new ScalaJS.c.jl_ClassCastException(
     instance + " is not an instance of " + classFullName);
```

Comment:
A very ugly and stupid solution would be to single out ClassCastException and never optimize it. Isn't it?